src/hooks/useCalculatedColumns.ts

```diff
@@ -148,38 +153,19 @@ export function useCalculatedColumns<R, SR>({
     let left = 0;
     let totalFrozenColumnWidth = 0;
     let templateColumns = '';
-    let allocatedWidth = 0;
-    let unassignedColumnsCount = 0;
 
     for (const column of columns) {
-      let width = getSpecifiedWidth(column, columnWidths, viewportWidth);
-
-      if (width === undefined) {
-        unassignedColumnsCount++;
-      } else {
+      let width = columnWidths.get(column.key) ?? flexColumnWidths.get(column.key) ?? column.width;
+      if (typeof width === 'number') {
         width = clampColumnWidth(width, column);
-        allocatedWidth += width;
-        columnMetrics.set(column, { width, left: 0 });
-      }
-    }
-
-    for (const column of columns) {
-      let width: number;
-      if (columnMetrics.has(column)) {
-        const columnMetric = columnMetrics.get(column)!;
-        columnMetric.left = left;
-        ({ width } = columnMetric);
       } else {
-        // avoid decimals as subpixel positioning can lead to cell borders not being displayed
-        const unallocatedWidth = viewportWidth - allocatedWidth;
-        const unallocatedColumnWidth = round(unallocatedWidth / unassignedColumnsCount);
-        width = clampColumnWidth(unallocatedColumnWidth, column);
-        allocatedWidth += width;
-        unassignedColumnsCount--;
-        columnMetrics.set(column, { width, left });
+        // This is a placeholder width so we can continue to use virtualization.
+        // The actual value is set after the column is rendered
+        width = DEFAULT_COLUMN_MIN_WIDTH;
```

Comment:
Shouldn't we do this instead?
```suggestion
        width = column.minWidth;
```